sklearn/model_selection/_validation.py

```diff
@@ -361,15 +358,25 @@ def _warn_about_fit_failures(results, error_score):
             for error, n in fit_errors_counter.items()
         )
 
-        some_fits_failed_message = (
-            f"\n{num_failed_fits} fits failed out of a total of {num_fits}.\n"
-            "The score on these train-test partitions for these parameters"
-            f" will be set to {error_score}.\n"
-            "If these failures are not expected, you can try to debug them "
-            "by setting error_score='raise'.\n\n"
-            f"Below are more details about the failures:\n{fit_errors_summary}"
-        )
-        warnings.warn(some_fits_failed_message, FitFailedWarning)
+        if num_failed_fits == num_fits:
+            all_fits_failed_message = (
+                f"\nAll the {num_fits} fits failed.\n"
+                "It is is very likely that your model is misconfigured.\n"
+                "You can try to debug the error by setting error_score='raise'.\n\n"
+                f"Below are more details about the failures:\n{fit_errors_summary}"
+            )
+            raise NotFittedError(all_fits_failed_message)
```

Comment:
As discussed IRL, maybe `NotFittedError` is not the best choice of exception type here. If all fit fail, it's very likely to be caused by an incompatible choice of parameters or a bad interaction with statiscal properties of the input data. So we could use `ValueError` instead.

Alternatively, we could record the type of the underlying exceptions, if it's uniformly the same for all fit calls and raise an exception of that type instead.

Reply:
Note that apparently, not raising `NotFittedError` here would cause a slight change of behavior in some edge case of multimetrics handling but this can probably considered a bug fix. Maybe @lesteve can give us more details about this point.

Reply:
Sorry I had not read https://github.com/scikit-learn/scikit-learn/pull/21026#issuecomment-929272762 before starting this thread ;)

Reply:
@thomasjpfan any opinion on the above? I have also the feeling that @jnothman has been gifted with uncommon abilities and judgement in the subtle art of choosing exception types.

Reply:
> Note that apparently, not raising NotFittedError here would cause a slight change of behavior in some edge case of multimetrics handling but this can probably considered a bug fix. Maybe @lesteve can give us more details about this point.

Quoting @thomasjpfan in https://github.com/scikit-learn/scikit-learn/pull/20619#pullrequestreview-717202617:

> With callable multimetric, at least one _fit_and_score has to succeed so that *SearchCV can create error_score dictionaries for the failed cases.

Before this PR, when all the fits failed in callable multimetric, the error was `NotFittedError`. This was done [here](https://github.com/scikit-learn/scikit-learn/pull/21026/files#diff-24fbe29b336ea0ad7ef67d54c362bfebd086d2367337526fb99dae4583891c2dL330-L331).

Reply:
Given the description we have for `NotFittedError`:

> Exception class to raise if estimator is used before fitting.

I do not think `NotFittedError` is suppose to be when fitting. I am in favor of changing the exception to `ValueError`.

Reply:
There seem to be consensus on this so I changed it to `ValueError`.